app/scripts/components/common/catalog/controls/hooks/use-catalog-view.ts

```diff
@@ -17,8 +19,45 @@ export function useCatalogView() {
   return {
     search,
     taxonomies,
-    setSearch,
-    setTaxonomies,
-    onAction
+    onCatalogViewAction,
   };
 }
+
+export function useCatalogViewQS() {
```

Comment:
I pushed a change that separates out QS-related states and actions into a separate hook. How do you think? @sandrahoang686 ? I think in this way, we don't have to trigger the initialization of atoms, and make it clearer what is controlling the query parameter.

Reply:
yep! I mentioned this in the description, it does look cleaner too, good with me! I opened this PR because I didn't want to merge to your branch directly in case you were cleaning so feel free to merge whenever comfortable